File: src/test/java/com/znsio/teswiz/screen/theapp/LoginScreen.java
package com.znsio.teswiz.screen.theapp;


public abstract class LoginScreen {

    public static LoginScreen get() {
        return com.znsio.teswiz.screen.ScreenRegistry.getScreen(LoginScreen.class);
    }

    public abstract LoginScreen enterLoginDetails(String username, String password);

    public abstract LoginScreen login();

    public abstract String getInvalidLoginError();

    public abstract LoginScreen dismissAlert();
}

File: src/test/resources/playwright/screens/theapp/login.screen.ts
import type { ScreenContext } from "../screen-context.ts";

const LOCATORS = {
  username: "#username",
  password: "#password",
  loginButtonRole: "button",
  loginButtonName: "Login",
  flashMessage: "#flash",
} as const;

export async function enterLoginDetails(
  screen: ScreenContext,
  username: string,
  password: string,
): Promise<void> {
  await screen.page.locator(LOCATORS.username).fill(username);
  await screen.page.locator(LOCATORS.password).fill(password);
}

export async function login(screen: ScreenContext): Promise<void> {
  await screen.page.getByRole(LOCATORS.loginButtonRole, { name: LOCATORS.loginButtonName }).click();
}

export async function getInvalidLoginError(screen: ScreenContext): Promise<string> {
  const invalidLoginError = await screen.page.locator(LOCATORS.flashMessage).textContent();
  return invalidLoginError?.trim() ?? "";
}

export async function dismissAlert(): Promise<void> {
}
